$OpenBSD: patch-src_demuxers_demux_4xm_c,v 1.1.2.1 2009/08/13 00:57:18 william Exp $

Upstream changeset 9757  7799748cc0f2:
Fix another possible 4xm demuxer integer overflow. 

--- src/demuxers/demux_4xm.c.orig	Tue Feb 10 12:32:23 2009
+++ src/demuxers/demux_4xm.c	Sat Jul 11 15:18:27 2009
@@ -190,9 +190,9 @@ static int open_fourxm_file(demux_fourxm_t *fourxm) {
         return 0;
       }
       const uint32_t current_track = _X_LE_32(&header[i + 8]);
-      if (current_track + 1 > fourxm->track_count) {
+      if (current_track >= fourxm->track_count) {
         fourxm->track_count = current_track + 1;
-        if (fourxm->track_count >= UINT_MAX / sizeof(audio_track_t)) {
+        if (!fourxm->track_count || fourxm->track_count >= UINT_MAX / sizeof(audio_track_t)) {
           free(header);
           return 0;
         }
